// @mistralai/mistralai/models/operations/librariesdocumentsdeletev1.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LibrariesDocumentsDeleteV1Request$ = exports.LibrariesDocumentsDeleteV1Request$outboundSchema = exports.LibrariesDocumentsDeleteV1Request$inboundSchema = void 0;
exports.librariesDocumentsDeleteV1RequestToJSON = librariesDocumentsDeleteV1RequestToJSON;
exports.librariesDocumentsDeleteV1RequestFromJSON = librariesDocumentsDeleteV1RequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.LibrariesDocumentsDeleteV1Request$inboundSchema = z.object({
    library_id: z.string(),
    document_id: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "library_id": "libraryId",
        "document_id": "documentId",
    });
});
/** @internal */
exports.LibrariesDocumentsDeleteV1Request$outboundSchema = z.object({
    libraryId: z.string(),
    documentId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        libraryId: "library_id",
        documentId: "document_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LibrariesDocumentsDeleteV1Request$;
(function (LibrariesDocumentsDeleteV1Request$) {
    /** @deprecated use `LibrariesDocumentsDeleteV1Request$inboundSchema` instead. */
    LibrariesDocumentsDeleteV1Request$.inboundSchema = exports.LibrariesDocumentsDeleteV1Request$inboundSchema;
    /** @deprecated use `LibrariesDocumentsDeleteV1Request$outboundSchema` instead. */
    LibrariesDocumentsDeleteV1Request$.outboundSchema = exports.LibrariesDocumentsDeleteV1Request$outboundSchema;
})(LibrariesDocumentsDeleteV1Request$ || (exports.LibrariesDocumentsDeleteV1Request$ = LibrariesDocumentsDeleteV1Request$ = {}));
function librariesDocumentsDeleteV1RequestToJSON(librariesDocumentsDeleteV1Request) {
    return JSON.stringify(exports.LibrariesDocumentsDeleteV1Request$outboundSchema.parse(librariesDocumentsDeleteV1Request));
}
function librariesDocumentsDeleteV1RequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.LibrariesDocumentsDeleteV1Request$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'LibrariesDocumentsDeleteV1Request' from JSON`);
}
