// @mistralai/mistralai/models/components/transcriptionstreamlanguage.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TranscriptionStreamLanguage$ = exports.TranscriptionStreamLanguage$outboundSchema = exports.TranscriptionStreamLanguage$inboundSchema = exports.TranscriptionStreamLanguageType$ = exports.TranscriptionStreamLanguageType$outboundSchema = exports.TranscriptionStreamLanguageType$inboundSchema = exports.TranscriptionStreamLanguageType = void 0;
exports.transcriptionStreamLanguageToJSON = transcriptionStreamLanguageToJSON;
exports.transcriptionStreamLanguageFromJSON = transcriptionStreamLanguageFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
exports.TranscriptionStreamLanguageType = {
    TranscriptionLanguage: "transcription.language",
};
/** @internal */
exports.TranscriptionStreamLanguageType$inboundSchema = z.nativeEnum(exports.TranscriptionStreamLanguageType);
/** @internal */
exports.TranscriptionStreamLanguageType$outboundSchema = exports.TranscriptionStreamLanguageType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TranscriptionStreamLanguageType$;
(function (TranscriptionStreamLanguageType$) {
    /** @deprecated use `TranscriptionStreamLanguageType$inboundSchema` instead. */
    TranscriptionStreamLanguageType$.inboundSchema = exports.TranscriptionStreamLanguageType$inboundSchema;
    /** @deprecated use `TranscriptionStreamLanguageType$outboundSchema` instead. */
    TranscriptionStreamLanguageType$.outboundSchema = exports.TranscriptionStreamLanguageType$outboundSchema;
})(TranscriptionStreamLanguageType$ || (exports.TranscriptionStreamLanguageType$ = TranscriptionStreamLanguageType$ = {}));
/** @internal */
exports.TranscriptionStreamLanguage$inboundSchema = (0, schemas_js_1.collectExtraKeys)(z.object({
    type: exports.TranscriptionStreamLanguageType$inboundSchema.default("transcription.language"),
    audio_language: z.string(),
}).catchall(z.any()), "additionalProperties", true).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "audio_language": "audioLanguage",
    });
});
/** @internal */
exports.TranscriptionStreamLanguage$outboundSchema = z.object({
    type: exports.TranscriptionStreamLanguageType$outboundSchema.default("transcription.language"),
    audioLanguage: z.string(),
    additionalProperties: z.record(z.any()),
}).transform((v) => {
    return {
        ...v.additionalProperties,
        ...(0, primitives_js_1.remap)(v, {
            audioLanguage: "audio_language",
            additionalProperties: null,
        }),
    };
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TranscriptionStreamLanguage$;
(function (TranscriptionStreamLanguage$) {
    /** @deprecated use `TranscriptionStreamLanguage$inboundSchema` instead. */
    TranscriptionStreamLanguage$.inboundSchema = exports.TranscriptionStreamLanguage$inboundSchema;
    /** @deprecated use `TranscriptionStreamLanguage$outboundSchema` instead. */
    TranscriptionStreamLanguage$.outboundSchema = exports.TranscriptionStreamLanguage$outboundSchema;
})(TranscriptionStreamLanguage$ || (exports.TranscriptionStreamLanguage$ = TranscriptionStreamLanguage$ = {}));
function transcriptionStreamLanguageToJSON(transcriptionStreamLanguage) {
    return JSON.stringify(exports.TranscriptionStreamLanguage$outboundSchema.parse(transcriptionStreamLanguage));
}
function transcriptionStreamLanguageFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.TranscriptionStreamLanguage$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TranscriptionStreamLanguage' from JSON`);
}
